src/execution_plan/optimizations/optimizations_util.c

```diff
@@ -0,0 +1,40 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "optimizations_util.h"
+#include "../../util/rax_extensions.h"
+
+void OptimizeUtils_MigrateFilterOp(ExecutionPlan *plan, OpBase *root, OpFilter *filter) {
+	rax *references = FilterTree_CollectModified(filter->filterTree);
+	OpBase *op = ExecutionPlan_LocateReferences(root, NULL, references);
+	ExecutionPlan_RemoveOp(plan, (OpBase *)filter);
+	ExecutionPlan_PushBelow(op, (OpBase *)filter);
```

Comment:
Since we will get here fairly often with filters that don't actually need to be migrated, we can save ourselves some unnecessary work with a construction like:
```
if(op != filter->op.children[0]) {
	ExecutionPlan_RemoveOp(plan, (OpBase *)filter);
	ExecutionPlan_PushBelow(op, (OpBase *)filter);
}
```

Reply:
fixed